sylvia/src/multitest.rs

```diff
@@ -1,62 +1,100 @@
 use std::cell::{Ref, RefCell, RefMut};
+use std::fmt::{Debug, Display};
 use std::marker::PhantomData;
+use std::ops::Deref;
 
-use cosmwasm_std::{Addr, BlockInfo, Coin};
-use cw_multi_test::Executor;
+use cosmwasm_std::{Addr, Api, Coin, CustomQuery, Empty, GovMsg, IbcMsg, IbcQuery, Storage};
+use cw_multi_test::{
+    BankKeeper, DistributionKeeper, Executor, FailingModule, Router, StakeKeeper, WasmKeeper,
+};
+use schemars::JsonSchema;
+use serde::de::DeserializeOwned;
 use serde::Serialize;
 
-#[derive(Default)]
-pub struct App {
-    pub app: RefCell<cw_multi_test::App>,
+pub struct App<MtApp> {
+    app: RefCell<MtApp>,
 }
 
-impl App {
-    pub fn new(app: cw_multi_test::App) -> Self {
-        Self {
-            app: RefCell::new(app),
-        }
+impl<MtApp> Default for App<MtApp>
+where
+    MtApp: Default,
+{
+    fn default() -> Self {
+        Self::new(MtApp::default())
     }
+}
 
-    pub fn app(&self) -> Ref<'_, cw_multi_test::App> {
-        Ref::map(self.app.borrow(), |app| app)
+impl<MtApp> Deref for App<MtApp> {
+    type Target = RefCell<MtApp>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.app
     }
+}
```

Comment:
I know I suggested it, but I was hoping we can do defer to `MtApp`, derefing to `RefCell<MtApp>` doesn't really make sense - better migrate functions to our `App`, giving the possibility to access `App` with `.app()` and `.app_mut()`.